rustls/src/client/hs.rs

```diff
@@ -160,6 +161,9 @@ pub(super) fn start_handshake(
         _ => None,
     };
 
+    let mut hello = ClientHelloDetails::new(extension_order_seed);
```

Comment:
It looks like this is the only callsite for `ClientHelloDetails::new()`? If so, I'd probably pass the `offered_group` in directly.